### PR TITLE
control-service: graphQL improved job executions sorting by end_time 

### LINKED ISSUE #440: GraphQL API - allow job executions sorting by end_time
**What is the feature request? What problem does it solve?**
As a Data Jobs owner I would like to list my jobs along with their latest 10 executions. In this way we will have an overview of the recent job executions status. Currently we have to retrieve all available executions to get the last ones.

**Use cases**
Dashboard that shows data jobs along with their latest 10 executions.

**Suggested solution**
Modify an existing GraphQL API by adding the ability to sort by `job.execution.end_time`.

1. Add new strategy in _JobFieldStrategyBy_
2. Extend _com.vmware.taurus.service.graphql.ExecutionDataFetcher_ to support filters. The filters should be performed at  the database level. (https://piotrminkowski.com/2020/07/31/an-advanced-guide-to-graphql-with-spring-boot/)

**Additional context**
Sample GraphQL query:
```
{{ baseUrl  }}/data-jobs/for-team/no-team-specified/jobs?query={
  jobs(pageNumber: 1, pageSize: 100, filter: [{property: "jobName", sort: ASC}]) {
    content {
      jobName
      config {
        team
        description
        schedule {
          scheduleCron
        }
      }
      deployments {
        enabled
        status
        executions(pageNumber: 1, pageSize: 10, filter: [{property: "endTime", sort: DESC}]) {
          id
          endTime
        }
      }
    }
  }
}
```
Sample GraphQL response:
```json
{
  "errors": [],
  "data": {
    "content": [
      {
        "jobName": "tms-staging-vdk-1633680894",
        "config": {
          "team": "taurus",
          "description": "",
          "schedule": {
            "scheduleCron": "* * * * *"
          }
        },
        "deployments": [
          {
            "enabled": true,
            "status": null,
            "executionsSucceeded": 100,
            "executionsFailed": 10,
            "executions": [
              {
                "id": "tms-staging-vdk-1633680894-1634638440",
                "endTime": "2021-10-19T10:16:47Z"
              },
              {
                "id": "tms-staging-vdk-1633680894-1634638500",
                "endTime": "2021-10-18T10:15:47Z"
              },
              {
                "id": "tms-staging-vdk-1633680894-1634638560",
                "endTime": "2021-10-17T10:14:47Z"
              }
            ]
          }
        ]
      }
    ],
    "totalItems": 0,
    "totalPages": 0
  }
}
```
**Acceptance criteria**
The GraphQL API should provide an ability to sort by `job.execution.end_time`.

### PR DESCRIPTION
**Pull request by mrMoZ1**
why: As a Data Jobs owner I would like to list my jobs along with 
their latest 10 executions. In this way we will have an overview 
of the recent job executions status. 

what: This feature was already implemented. Improved query
efficiency slightly and added unit tests. Closes #440

testing: Added unit test. Manually ran queries against a locally
running control-service and made sure everything is in order.

Signed-off-by: Momchil Zhivkov <mzhivkov@vmware.com>

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.UUID;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest(classes = ServiceApp.class)
+@AutoConfigureMockMvc
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class GraphQLJobExecutionsSortByEndTimeIT {
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java:30]: This test looks like integration test. What do you think? @mrMoZ1, @tpalashki

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java:30]: That was the idea. Is this not the correct place? The test makes real calls to the graphQl and runs quickly it also exercises the sort and pagination logic.

Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java:30]: Yes, it should be under the `src/integration-test`.

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java:30]: It depends on a method in the com.vmware.taurus.RepositoryUtil class in the testing classes which means that I will have to copy/paste this method in the IT test directory. I also did see that we have other IT tests in the same package so decided to put it there. Should I move it to the integration-test directory?

Comment (mivanov1988) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java:30]: You can easily replace it with jobExecutionRepository.save().

Comment (mrMoZ1) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/GraphQLJobExecutionsSortByEndTimeIT.java]: moved test to integration-test directory

**Review by mivanov1988 — APPROVED** [review]

**Pull request merged**